assets/js/components/ResetButton.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * ResetButton component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Fragment, useState, useEffect, useCallback, createInterpolateElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { clearWebStorage } from '../util';
+import Dialog from './dialog';
+import Modal from './Modal';
+import Link from './Link';
+import { STORE_NAME as CORE_SITE } from '../googlesitekit/datastore/site/constants';
+
+const { useSelect, useDispatch } = Data;
+
+function ResetButton( { children } ) {
+	const postResetURL = useSelect( ( select ) => select( CORE_SITE ).getAdminURL( 'googlesitekit-splash', { notification: 'reset_success' } ) );
+
+	const [ dialogActive, setDialogActive ] = useState( false );
+	// const postResetURL = addQueryArgs( splashURL,  );
+
+	const handleCloseModal = useCallback( ( event ) => {
```

Comment:
Let's move the definition of this function into the `useEffect` below since that's the only place it's used.